daml-assistant/src/DAML/Assistant/Install/Github.hs

```diff
@@ -0,0 +1,112 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE OverloadedStrings #-}
+
+-- | Discover releases from the digital-asset/daml github.
+module DAML.Assistant.Install.Github
+    where
+
+import DAML.Assistant.Types
+import DAML.Assistant.Util
+import Data.Aeson
+import Network.HTTP.Simple
+import Control.Exception.Safe
+import Control.Monad
+import Data.List
+import Data.Either.Extra
+import qualified System.Info
+import qualified Data.Text as T
+
+newtype Tag = Tag Text deriving (Eq, Show, FromJSON)
+newtype AssetName = AssetName { unAssetName :: Text } deriving (Eq, Show, FromJSON)
+
+data Release = Release
+    { releaseTag          :: Tag
+    , releaseIsPrerelease :: Bool
+    , releaseAssets       :: [Asset]
+    } deriving (Eq, Show)
+
+instance FromJSON Release where
+    parseJSON = withObject "Release" $ \r ->
+        Release
+        <$> r .: "tag_name"
+        <*> r .: "prerelease"
+        <*> r .: "assets"
+
+data Asset = Asset
+    { assetName :: AssetName
+    , assetDownloadURL :: InstallURL
+    } deriving (Eq, Show)
+
+instance FromJSON Asset where
+    parseJSON = withObject "Asset" $ \r ->
+        Asset
+        <$> r .: "name"
+        <*> r .: "browser_download_url"
+
+versionToTag :: SdkVersion -> Tag
```

Comment:
If Tag must be isomorphic to SdkVersion, why define Tag at all? Why not just pass around SdkVersion?

Reply:
They aren't isomorphic. Tag represents git tags, only some of which represent valid versions.